dataflow/flex-templates/streaming_beam_sql/README.md

```diff
@@ -0,0 +1,315 @@
+
+1. Clone the
+    [`java-docs-samples` repository](https://github.com/GoogleCloudPlatform/java-docs-samples)
+    and navigate to the code sample.
+
+    ```sh
+    git clone https://github.com/GoogleCloudPlatform/java-docs-samples.git
+    cd java-docs-samples/dataflow/flex-templates/beam_sql
+    ```
+
+## Pub/Sub to BigQuery with Beam SQL sample
+
+This sample shows how to deploy an Apache Beam streaming pipeline that reads
+[JSON encoded](https://www.w3schools.com/whatis/whatis_json.asp) messages from
+[Pub/Sub](https://cloud.google.com/pubsub), uses
+[Beam SQL](https://beam.apache.org/documentation/dsls/sql/overview/)
+to transform the message data, and writes the results to a
+[BigQuery](https://cloud.google.com/bigquery) table.
+
+* [Dockerfile](Dockerfile)
+* [StreamingBeamSQL.java](src/main/java/org/apache/beam/samples/StreamingBeamSQL.java)
+* [pom.xml](pom.xml)
+* [metadata.json](metadata.json)
+
+### Building a container image
+
+> <details><summary>
+> <i>(Optional)</i> Run the Apache Beam pipeline locally for development.
+> <i>(Click to expand)</i>
+> </summary>
+>
+> ```sh
+> mvn compile exec:java \
+>   -Dexec.mainClass=org.apache.beam.samples.StreamingBeamSQL \
+>   -Dexec.args="\
+>     --inputSubscription=$SUBSCRIPTION \
+>     --outputTable=$PROJECT:$DATASET.$TABLE \
+>     --tempLocation=gs://$BUCKET/samples/dataflow/temp"
+> ```
+>
+> </details>
+
+Build the Java project into an
+[*Uber JAR* file](https://maven.apache.org/plugins/maven-shade-plugin/).
+
+```sh
+# Build and package the application as an uber-jar file.
+mvn clean package
+
+# (Optional) Note the size of the uber-jar file compared to the original.
+ls -lh target/*.jar
+```
+
+This *Uber JAR* file has all the dependencies embedded so it.
+You can run this file as a standalone application with no external
+dependencies on other libraries.
+
+Now, we build the
+[Docker](https://docs.docker.com/engine/docker-overview/)
+image for the Apache Beam pipeline.
+We are using
+[Cloud Build](https://cloud.google.com/cloud-build)
+so we don't need a local installation of Docker.
+
+> *Note:* You can speed up subsequent builds with
+> [Kaniko cache](https://cloud.google.com/cloud-build/docs/kaniko-cache)
+> in Cloud Build.
+>
+> ```sh
+> # (Optional) Enable to use Kaniko cache by default.
+> gcloud config set builds/use_kaniko True
+> ```
+
+Cloud Build allows you to
+[build a Docker image using a `Dockerfile`](https://cloud.google.com/cloud-build/docs/quickstart-docker#build_using_dockerfile).
+and saves it into
+[Container Registry](https://cloud.google.com/container-registry/),
+where the image is accessible to other Google Cloud products.
+
+```sh
+export TEMPLATE_IMAGE="$PROJECT/samples/dataflow/streaming-beam-sql:latest"
+
+# Build the image into Container Registry, this is roughly equivalent to:
+#   gcloud auth configure-docker
+#   docker image build -t $TEMPLATE_IMAGE .
+#   docker push $TEMPLATE_IMAGE
+gcloud builds submit --tag "gcr.io/$TEMPLATE_IMAGE" .
+```
```

Comment:
So this is interesting w/ that .cloudignore above -- I presume you've verified that things work.